File: src/main/python/ch03/sec05/demo_adjust_model.py
# -*- encoding: utf-8 -*-    
"""
@Author     :   zYx.Tom
@Contact    :   526614962@qq.com
@site       :   https://zhuyuanxiang.github.io
---------------------------
@Software   :   PyCharm
@Project    :   Hanlp-Books-Examples
@File       :   demo_adjust_model.py
@Version    :   v0.1
@Time       :   2020-12-25 10:03
@License    :   (C)Copyright 2018-2020, zYx.Tom
@Reference  :   
@Desc       :   
@理解：
"""
from pyhanlp import HanLP

from ch03.msr import msr_model
from ch03.sec03.demo_ngram_segment import load_bigram
from preamble import CoreDictionary
from tools import *


# ----------------------------------------------------------------------
def main():
    segment = load_bigram(model_path=msr_model, verbose=True, ret_viterbi=True)
    assert CoreDictionary.contains("管道")
    text = "北京输气管道工程"
    HanLP.Config.enableDebug()
    print(segment.seg(text))
    pass


if __name__ == '__main__':
    # Note: 运行后可能会影响到环境的编码设置，导致 Console 输出为乱码，重启一次环境就好了
    main()
    beep_end()

File: src/main/java/ch03/sec05/DemoAdjustModel.java
/*
 * @author zYx.Tom
 * @email zhuyuanxiang@gmail.com
 * @date 2020-12-25 9:58
 */

package ch03.sec05;

import com.hankcs.hanlp.HanLP;
import com.hankcs.hanlp.dictionary.CoreDictionary;
import com.hankcs.hanlp.seg.Segment;

import static ch03.MSR.MSR_MODEL_PATH;
import static ch03.sec03.DemoNgramSegment.loadBigram;

/**
 * 《自然语言处理入门》
 * 3.5.3 调整模型
 */
public class DemoAdjustModel {
    public static void main(String[] args) {
        Segment segment = loadBigram(MSR_MODEL_PATH, false, false);
        assert CoreDictionary.contains("管道");
        String text = "北京输气管道工程";
        HanLP.Config.enableDebug();
        System.out.println(segment.seg(text));
    }
}
